yt/visualization/line_plot.py

```diff
@@ -31,6 +31,12 @@
     linear_transform, \
     invalidate_plot
 
+class LineObject(object):
+    def __init__(self, start_point, end_point, ds, label=None):
+        self.start_point = _validate_point(start_point, ds, start=True)
+        self.end_point = _validate_point(end_point, ds)
+        self.label = label
```

Comment:
I'm actually not sure, for this reason--once the dust settles on this, it'd be good (maybe in a separate PR, or maybe not) to make `LineObject` a thing that one can query just like `FixedResolutionBuffer`, so that when you make one, you can do something like this:

```python
lo = LineObject(...)
lo["density"]
```